modules/kinesis/src/it/scala/com/snowplowanalytics/snowplow/enrich/kinesis/EnrichKinesisSpec.scala

```diff
@@ -46,11 +50,24 @@ class EnrichKinesisSpec extends Specification with AfterAll with CatsIO {
       val nbGood = 100l
       val nbBad = 10l
 
+      def containsEnrichmentsContexts(event: Event): Boolean =
```

Comment:
> What do you think about leaving 'counting' spec as is and add separate test for checking enrichments?

Yes agreed, I'll separate them.

> And to make it easier, we could create only one input collector payload (not generated one) and assert it.

Given that we don't assert the output (see below), I think that it's good to keep on generating the payloads on the fly, so that there is a chance that random values with edge cases might fight a bug.

> I think it would be also cool to check entire content of derived contexts.

Please see https://github.com/snowplow/enrich/pull/711#issuecomment-1325357345

Reply:
Yes, we should use generated payloads. We could also use created manually. Why not both?

We can have some 'uber spec' with generated input just to smoke test and see the count good/bad is correct. But I think we can also have other specs focusing on more detailed cases, like output contexts. Is it the concern that if we have too many tests, then whole test workflow would be very slow?

Reply:
> We could also use created manually. Why not both?

What's the added value of adding manual payloads to generated ones for a test ?

> We can have some 'uber spec' with generated input just to smoke test and see the count good/bad is correct. But I think we can also have other specs focusing on more detailed cases, like output contexts.

As I see it, the purpose of the integration tests is precisely to smoke test the runtime, and is not to cover the whole enrichments codebase. For that we have the unit tests (some of them [also communicating with outside services](https://github.com/snowplow/enrich/tree/master/integration-tests/common), that should be replaced with `testcontainers` when we get some time). That's why I think that this is not the right place for checking the content of the contexts added by enrichments.

> Is it the concern that if we have too many tests, then whole test workflow would be very slow?

It's my main concern if we want to run one Enrich instance per enrichment indeed.

Reply:
>What's the added value of adding manual payloads to generated ones for a test ?

It's easier to assert output details I think? But it really depends what kind of details we want to check! If we don't want to dig deep here in the output, generated input is perfectly fine. 

I also thought about those tests as a kind of sandbox/playground for us. If you want to run enrich with some custom configuration + manually created data, you can go here, create your scenario and get quickly a feedback. I just really like this setup and I imagine that would be the first place I go if I want to test/check how something in enrich works! That's why I care about flexibility and not fixing on `we use only generated stuff here`

> For that we have the unit tests  

So... if we communicate with outside services + use testcontainers, what will be the difference between those unit tests and this integration setup? I am often not able to tell what's unit/integration, now I mostly think about tests in terms of fast/slow ;)

Reply:
> If we don't want to dig deep here in the output, generated input is perfectly fine.

That's the case for now. It will be very easy to add fixed payloads if we need to.

> I just really like this setup and I imagine that would be the first place I go if I want to test/check how something in enrich works!

I'm very pleased to read that! I feel the same, and I look forward to having that for more apps (especially loaders).

> if we communicate with outside services + use testcontainers, what will be the difference between those unit tests and this integration setup?

The way I see it, the big difference is that unit tests are for [white box testing](https://www.guru99.com/white-box-testing.html) while the integration tests are for [black box testing](https://www.guru99.com/black-box-testing.html). Here we test the app as a black box, we run the same Docker image as in production, we use config files, we send events to the input, we read the output.. The main purpose is kind of smoke testing the features. I don't think that there is much added-value in trying to cover what the unit tests already cover.